streams/src/test/java/org/apache/kafka/streams/integration/EosBetaUpgradeIntegrationTest.java

```diff
@@ -993,6 +986,17 @@ private void waitForStateTransition(final List<KeyValue<KafkaStreams.State, Kafk
         );
     }
 
+    private void waitForStateTransitionContains(final List<KeyValue<KafkaStreams.State, KafkaStreams.State>> observed,
+                                        final List<KeyValue<KafkaStreams.State, KafkaStreams.State>> expected)
+            throws Exception {
+
+        waitForCondition(
+            () -> observed.containsAll(expected),
+            MAX_WAIT_TIME_MS,
+            () -> "Client did not have the expected state transition on time. Observers transitions: " + observed
```

Comment:
Can we add the expected transitions, too? Easier to debug if the test fails.

Reply:
sure that is fine